packages/nodes-base/nodes/Gong/GenericFunctions.ts

```diff
@@ -117,79 +115,95 @@ export const getCursorPaginator = (rootProperty: string | null = null) => {
 		let nextCursor: string | undefined = undefined;
 		const returnAll = this.getNodeParameter('returnAll', true) as boolean;
 
-		const extractItems = (page: INodeExecutionData) => {
-			let items: IDataObject[] = [page.json];
-			if (rootProperty) {
-				const paths = toPath(rootProperty);
-				for (const path of paths) {
-					items = items.flatMap((x) => get(x, path)) as IDataObject[];
-				}
-			}
-			if (items.length > 0) {
-				executions = executions.concat(items.map((item) => ({ json: item })));
-			}
-		};
-
 		do {
 			(requestOptions.options.body as IDataObject).cursor = nextCursor;
 			responseData = await this.makeRoutingRequest(requestOptions);
 			const lastItem = responseData[responseData.length - 1].json;
 			nextCursor = (lastItem.records as IDataObject)?.cursor as string | undefined;
-			responseData.forEach(extractItems);
+			for (const page of responseData) {
+				executions = executions.concat(extractItems(page));
+			}
 		} while (returnAll && nextCursor);
 
 		return executions;
 	};
 };
 
-export const getCursorPaginatorCalls = () => {
-	return async function cursorPagination(
-		this: IExecutePaginationFunctions,
-		requestOptions: DeclarativeRestApiSettings.ResultOptions,
-	): Promise<INodeExecutionData[]> {
-		let executions: INodeExecutionData[] = [];
-		let responseData: INodeExecutionData[];
-		let nextCursor: string | undefined = undefined;
-		const returnAll = this.getNodeParameter('returnAll', true) as boolean;
+const extractCalls = (page: INodeExecutionData): INodeExecutionData[] => {
+	const items: IDataObject[] = get(page.json, 'calls') as IDataObject[];
+	return items
+		.filter((item) => item?.metaData)
```

Comment:
since items can be empty, this is throwing an error when it shouldn't 
```suggestion
		?.filter((item) => item?.metaData)
```